jupyterlab_automl/src/service/kernel_model.ts

```diff
@@ -0,0 +1,111 @@
+import { IClientSession } from '@jupyterlab/apputils';
+
+import { Kernel, KernelMessage } from '@jupyterlab/services';
+
+import { ISignal, Signal } from '@phosphor/signaling';
+
+export class KernelModel {
+  constructor(
+    session: IClientSession,
+    refresh: () => void,
+    onError: (error: string) => void
+  ) {
+    this._session = session;
+    this._refresh = refresh;
+    this._onError = onError;
+  }
+
+  get future(): Kernel.IFuture<
+    KernelMessage.IExecuteRequestMsg,
+    KernelMessage.IExecuteReplyMsg
+  > | null {
+    return this._future;
+  }
+
+  set future(
+    value: Kernel.IFuture<
+      KernelMessage.IExecuteRequestMsg,
+      KernelMessage.IExecuteReplyMsg
+    > | null
+  ) {
+    this._future = value;
+    if (!value) {
+      return;
+    }
+    value.onIOPub = this._onIOPub;
+  }
+
+  get session(): IClientSession {
+    return this._session;
+  }
+
+  get output(): string | null {
+    return this._output;
+  }
+
+  get name(): string | null {
+    return this._name;
+  }
+
+  get refresh(): () => void | null {
+    return this._refresh;
+  }
+
+  get onError(): (error: string) => void | null {
+    return this._onError;
+  }
+
+  get receivedData(): ISignal<KernelModel, void> {
+    return this._receivedData;
+  }
+
+  get receivedError(): ISignal<KernelModel, string> {
+    return this._receivedError;
+  }
+
+  createCSV(name: string, df: string) {
+    this._name = name;
+    if (!this._session || !this._session.kernel) {
+      return;
+    }
+    const path = './' + df + '.csv';
+    const run =
+      "import base64;df.to_csv('" +
+      path +
+      "', index=False);base64.b64encode(open('" +
+      path +
+      "', 'r').read().encode('utf8')).decode('utf-8')";
```

Comment:
- this goes into the kernel and turns the dataframe to a csv and converts that to bytes to create a dataset
- the name of the csv file is the name of the variable the user put in the import data form
- should this file be deleted from the local file system once the bytes have been uploaded?